sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/utils.scala

```diff
@@ -70,7 +70,6 @@ object Utils {
     // Check if we can use TungstenAggregate.
     val usesTungstenAggregate =
       child.sqlContext.conf.unsafeEnabled &&
-      aggregateExpressions.forall(_.aggregateFunction.isInstanceOf[DeclarativeAggregate]) &&
```

Comment:
Whoops, good catch. I removed this from `planAggregateWithOneDistinct` and it led to a failure in `single distinct column set`:

```
[info]   == Results ==
[info]   !== Correct Answer - 4 ==                                                        == Spark Answer - 4 ==
[info]   ![100.0,0.3333333333333333,1.0,2,99.33333333333333,10.0,2.0]                     [100.0,0.3333333333333333,1.0,2,101.0,10.0,2.0]
[info]   ![110.0,10.0,20.0,null,109.0,11.0,30.0]                                          [110.0,10.0,20.0,null,120.0,11.0,30.0]
[info]   ![120.0,23.333333333333332,-3.3333333333333335,1,122.33333333333333,12.0,20.0]   [120.0,23.333333333333332,-3.3333333333333335,1,96.66666666666667,12.0,20.0]
[info]   ![null,null,3.0,3,null,null,null]                                                [null,null,3.0,3,103.0,null,null] (AggregationQuerySuite.scala:690)
```

Based on the failure mode here, I suspect that another one of the buffer offsets needs to be reset in response to the sort fallback.